### PR TITLE
Soil and moisture types

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Proposal for defaulting ground conductivity and diffusivity based on optional HPXML fields SoilType and MoistureType.

image

Source comparisons:

SoilType | MoistureType | [NLM](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4813881) |   | GLHEPro |   | Existing |  
-- | -- | -- | -- | -- | -- | -- | --
  |   | Conductivity | Diffusitivity | Conductivity | Diffusitivity | Conductivity | Diffusitivity
loam | dry/wet/mixed | 1.213 | 0.035 |   |   |   |  
gravel | dry | 0.231 | 0.009 |   |   |   |  
gravel | wet | 1.039 | 0.029 |   |   |   |  
sand | dry | 0.231 | 0.009 | 0.199 | 0.0111 |   |  
sand | wet | 1.386 | 0.032 |   |   |   |  
sand | mixed |   |   | 0.5 | 0.02 |   |  
clay | dry | 0.288 | 0.012 | 0.5174 | 0.0207 |   |  
clay | wet | 0.982 | 0.019 | 1.4 | 0.0351 |   |  
clay | mixed |   |   | 0.75 | 0.0249 |   |  
silt | dry | 0.288 | 0.012 |   |   |   |  
silt | wet | 0.982 | 0.019 |   |   |   |  
unknown | dry/wet/mixed |   |   |   |   | 1 | 0.0208

Adds new BuildResidentialHPXML measure arguments:
* `site_soil_type`
* `site_moisture_type`

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [x] Schematron validator (`EPvalidator.xml`) has been updated
- [x] Sample files have been added/updated (via `tasks.rb`)
- [x] Tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests` and/or `workflow/tests/hpxml_translator_test.rb`)
- [x] Documentation has been updated
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: docs/source/workflow_inputs.rst [hunk 1]
```diff
@@ -546,15 +546,25 @@ Site information is entered in ``/HPXML/Building/BuildingDetails/BuildingSummary
...
   .. [#] Diffusivity used for ground source heat pumps.
+  .. [#] MoistureType choices are "mixed", "wet", or "dry".  
+  .. [#] | If Conductivity and extension/Diffusivity not provided, defaults based on SoilType/MoistureType:
+         | - **unknown**, **mixed/wet/dry**: 1.0 Btu/hr-ft-F and 0.0208 ft2/hr
+         | - **sand**, **mixed**: 0.5 Btu/hr-ft-F and 0.02 ft2/hr
+         | - **sand**, **wet**: Not supported
+         | - **sand**, **dry**: 0.199 Btu/hr-ft-F and 0.0111 ft2/hr
+         | - **clay**, **mixed**: 0.75 Btu/hr-ft-F and 0.0249 ft2/hr
+         | - **clay**, **wet**: 1.4 Btu/hr-ft-F and 0.0351 ft2/hr
+         | - **clay**, **dry**: 0.5174 Btu/hr-ft-F and 0.0207 ft2/hr
```
Comment (shorowit) [inline docs/source/workflow_inputs.rst:567]: Where did these values come from? Also, if you haven't seen it, I had dug up various values in the literature here: https://github.com/NREL/OpenStudio-HPXML/issues/1050

Comment (joseph-robertson) [reply docs/source/workflow_inputs.rst:567]: From here: https://github.com/NREL/OpenStudio-HPXML/pull/1391#issuecomment-1664600014

**Comment by joseph-robertson** [comment]
From the following table, we can get:
* Loam (moisture?)
* Dry sand
* Water saturated sand
* Dry clay/silt

image

**Comment by shorowit** [comment]
Note that the table above is cut off; there are two more rows that you can find in https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4813881/table/sensors-16-00306-t001/?report=objectonly.

**Comment by joseph-robertson** [comment]
Plans for ResStock? Map climate zones to soil/moisture types?

**Comment by joseph-robertson** [comment]
Allow using "mixed" by just averaging "wet" and "dry".

**Pull request merged**